src/mode/modeHandler.ts

```diff
@@ -129,50 +129,63 @@ export class VimSettings {
   useCtrlKeys     = false;
 }
 
+export enum SearchDirection {
+  Forward = 1,
+  Backward = -1
+}
+
 export class SearchState {
+  private static readonly MAX_SEARCH_RANGES = 1000;
+
   /**
    * Every range in the document that matches the search string.
    */
-  public matchRanges: vscode.Range[] = [];
+  private _matchRanges: vscode.Range[] = [];
+  public get matchRanges(): vscode.Range[] {
```

Comment:
Above comment only works for https://github.com/VSCodeVim/Vim/pull/494/files#diff-a7723f32fe6b05e135838f84e5c23c0fR176 . Even though `matchRanges` is readonly, you can't re-initialize it but you can always change its content, like pushing more elements to it.